sktime/transformers/series_as_features/catch22_features.py

```diff
@@ -0,0 +1,159 @@
+""" catch22 features
+A transformer for the catch22 features
+"""
+
+__author__ = "Matthew Middlehurst"
+__all__ = ["Catch22"]
+
+import numpy as np
+import pandas as pd
+import catch22
+from sktime.transformers.series_as_features.base import \
+    BaseSeriesAsFeaturesTransformer
+from sktime.utils.data_container import tabularize
+from sktime.utils.validation.series_as_features import check_X
+
+
+class Catch22(BaseSeriesAsFeaturesTransformer):
+    """ Canonical Time-series Characteristics (catch22)
+
+    @article{lubba2019catch22,
+        title={catch22: CAnonical Time-series CHaracteristics},
+        author={Lubba, Carl H and Sethi, Sarab S and Knaute, Philip and
+                Schultz, Simon R and Fulcher, Ben D and Jones, Nick S},
+        journal={Data Mining and Knowledge Discovery},
+        volume={33},
+        number={6},
+        pages={1821--1852},
+        year={2019},
+        publisher={Springer}
+    }
+
+    Overview: Input n series length m
+    Transforms series into the 22 catch22 features extracted from the hctsa
+    toolbox.
+
+    Fulcher, B. D., & Jones, N. S. (2017). hctsa: A computational framework
+    for automated time-series phenotyping using massive feature extraction.
+    Cell systems, 5(5), 527-531.
+
+    Fulcher, B. D., Little, M. A., & Jones, N. S. (2013). Highly comparative
+    time-series analysis: the empirical structure of time series and their
+    methods. Journal of the Royal Society Interface, 10(83), 20130048.
+
+    catch22 package implementations:
+    https://github.com/chlubba/catch22
+
+    For the Java version, see
+    https://github.com/uea-machine-learning/tsml/blob/master/src/main/java
+    /tsml/transformers/Catch22.java
+
+    """
+    def __init__(self):
+        super(Catch22, self).__init__()
+
+    def transform(self, X, y=None):
+        """transforms data into the catch22 features
+
+            Parameters
+            ----------
+            X : nested pandas DataFrame of shape [n_instances, 1]
+                Nested dataframe with univariate time-series in cells.
+            y : array-like, shape = [n_instances] The class labels.
+
+            Returns
+            -------
+            Pandas dataframe containing 22 features for each input series
+        """
+        self.check_is_fitted()
+        X = check_X(X, enforce_univariate=False)
+        X = tabularize(X, return_array=True)
+
+        n_instances = X.shape[0]
+
+        c22_list = []
+        for i in range(n_instances):
+            series = X[i, :]
+            c22_dict = catch22.catch22_all(series)
+            c22_list.append(c22_dict['values'])
+
+        return pd.DataFrame(c22_list)
+
+    def transform_single_feature(self, X, feature):
```

Comment:
do we need this? If we use this somewhere I'd make it private (i.e. leading underscore), otherwise we can handle this with the ColumnTransformer, no?

Reply:
By default the transform returns calculates and returns all 22 features. What that method does is returns 1 out of the 22 based on the feature parameter. This is useful for i.e. subsampling the catch22 features, if you want a random 10 then calculating the other 12 is a waste of time. I can make it private.